### PR TITLE
V4 resources

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

File: gnomad_qc/v4/resources/sample_qc.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            release: TableResource(
+                _get_ancestry_pca_ht_path(
+                    "loadings", release, include_unreleasable_samples
+                )
+            )
+            for release in VERSIONS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:86]:
```suggestion
            version: TableResource(
                _get_ancestry_pca_ht_path(
                    "loadings", version, include_unreleasable_samples
                )
            )
            for version in VERSIONS
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            release: TableResource(
+                _get_ancestry_pca_ht_path(
+                    "scores", release, include_unreleasable_samples
+                )
+            )
+            for release in VERSIONS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:108]:
```suggestion
            version: TableResource(
                _get_ancestry_pca_ht_path(
                    "scores", version, include_unreleasable_samples
                )
            )
            for version in VERSIONS
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            release: TableResource(
+                _get_ancestry_pca_ht_path(
+                    "eigenvalues", release, include_unreleasable_samples
+                )
+            )
+            for release in VERSIONS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:130]:
```suggestion
            version: TableResource(
                _get_ancestry_pca_ht_path(
                    "eigenvalues", version, include_unreleasable_samples
                )
            )
            for version in VERSIONS
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+# TODO: What file will this be for v4 comparison? Previously QC Sites (gnomAD v2 QC sites, lifted over)
+gnomad_v2_qc_sites = TableResource(
+    "gs://gnomad-public/resources/grch38/gnomad_v2_qc_sites_b38.ht"
+)
+
+# Dense MT of samples at QC sites
+qc = VersionedMatrixTableResource(
+    CURRENT_VERSION,
+    {
+        release: MatrixTableResource(
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:161]:
```suggestion
        version: MatrixTableResource(
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+            # TODO: What set this path to?
+            f"gs://gnomad/sample_qc/mt/genomes_v{release}/gnomad_v{release}_qc_mt_v3_sites_dense.mt"
+        )
+        for release in VERSIONS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:165]:
```suggestion
        for version in VERSIONS
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+)
+
+# PC relate PCA scores
+pc_relate_pca_scores = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        release: TableResource(
+            f"{get_sample_qc_root(release)}/gnomad_v{release}_qc_mt_v3_sites_pc_scores.ht"
+        )
+        for release in VERSIONS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:176]:
```suggestion
        version: TableResource(
            f"{get_sample_qc_root(version)}/gnomad_v{version}_pc_scores.ht"
        )
        for version in VERSIONS
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+)
+
+# PC relate results
+relatedness = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        release: TableResource(
+            f"{get_sample_qc_root(release)}/gnomad_v{release}_qc_mt_v3_sites_relatedness.ht"
+        )
+        for release in VERSIONS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:187]:
```suggestion
        version: TableResource(
           f"{get_sample_qc_root(version)}/gnomad_v{version}_relatedness.ht"
        )
        for version in VERSIONS
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 8]
```diff
@@ -0,0 +1,375 @@
+import hail as hl
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:1]: Note: Part way through this file I stopped changing `release` to `version`, can you change them all?

File: gnomad_qc/v4/resources/sample_qc.py [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+    """
+    Path to tab delimited file indicating inferred sample populations.
+
+    :param version: gnomAD Version
+    :return: String path to sample populations
+    """
+    return f"gs://gnomad/v{version}/sample_qc/temp/exomes/gnomad_v{version}_RF_pop_assignments.txt.gz"
+
+
+# TODO: Should this still be a "temp" directory?
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:257]: Good call, nope

File: gnomad_qc/v4/resources/sample_qc.py [hunk 10]
```diff
@@ -0,0 +1,375 @@
...
+
+
+def pop_tsv_path(version: str = CURRENT_VERSION) -> str:
+    """
+    Path to tab delimited file indicating inferred sample populations.
+
+    :param version: gnomAD Version
+    :return: String path to sample populations
+    """
+    return f"gs://gnomad/v{version}/sample_qc/temp/exomes/gnomad_v{version}_RF_pop_assignments.txt.gz"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:254]:
```suggestion
    return f"gs://gnomad/v{version}/sample_qc/exomes/gnomad_v{version}_RF_pop_assignments.txt.gz"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 11]
```diff
@@ -0,0 +1,375 @@
...
+
+
+def pop_rf_path(version: str = CURRENT_VERSION) -> str:
+    """
+    Path to RF model used for inferring sample populations.
+
+    :param version: gnomAD Version
+    :return: String path to sample pop RF model
+    """
+    return f"gs://gnomad/v{version}/sample_qc/temp/exomes/gnomad_v{version}_pop.RF_fit.pickle"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:267]:
```suggestion
    return f"gs://gnomad/v{version}/sample_qc/exomes/gnomad_v{version}_pop.RF_fit.pickle"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 12]
```diff
@@ -0,0 +1,375 @@
...
+    {
+        release: TableResource(
+            f"{get_sample_qc_root(release)}/gnomad_v{release}_hard_filtered_samples.ht"
+        )
+        for release in VERSIONS
+    },
+)
+
+# Results of running population-based metrics filtering
+# For v4 might not use this (regresed metrics used instead)
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:289]: Yeah we might not use it, but we will probably still calculate it to look at

File: gnomad_qc/v4/resources/sample_qc.py [hunk 13]
```diff
@@ -0,0 +1,375 @@
...
+    CURRENT_VERSION,
+    {
+        release: TableResource(
+            f"{get_sample_qc_root(release)}/gnomad_v{release}_release_samples_ranking.ht"
+        )
+        for release in VERSIONS
+    },
+)
+
+# Picard metrics
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:333]: We will probably change this to just have the picard metrics on the project metadata HT, so maybe move this to the meta.py resource file

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
Thank you so much for putting this in Kristen!

File: gnomad_qc/v4/resources/constants.py [hunk 14]
```diff
@@ -0,0 +1,5 @@
+VERSIONS = ["4"]
+CURRENT_VERSION = "4.0"
```
[the remark this answers is not in the document]
Comment (klaricch) [reply gnomad_qc/v4/resources/constants.py:2]: i like 4.0 -- seems like a more consistent format with later versions when we increment

File: gnomad_qc/v4/resources/sample_qc.py [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# PC relate PCA scores
+pc_relate_pca_scores = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_pc_scores.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:167]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.pc_scores.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# PC relate results
+relatedness = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_relatedness.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:178]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.relatedness.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Sex imputation results
+sex = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_sex.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:189]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.sex.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Samples to drop for PCA due to them being related
+pca_related_samples_to_drop = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_related_samples_to_drop_for_pca.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:200]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.related_samples_to_drop_for_pca.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Related samples to drop for release
+release_related_samples_to_drop = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_related_release_samples_to_drop.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:211]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.related_release_samples_to_drop.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 20]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Number of clinvar variants per sample
+sample_clinvar_count = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_clinvar.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:222]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.clinvar.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 21]
```diff
@@ -0,0 +1,354 @@
...
+
+
+def pop_rf_path(version: str = CURRENT_VERSION) -> str:
+    """
+    Path to RF model used for inferring sample populations.
+
+    :param version: gnomAD Version
+    :return: String path to sample pop RF model
+    """
+    return f"gs://gnomad/v{version}/sample_qc/exomes/gnomad_exomes_v{version}_pop.RF_fit.pickle"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:257]:
```suggestion
    return f"gs://gnomad/v{version}/sample_qc/exomes/gnomad.exomes.v{version}.pop.RF_fit.pickle"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 22]
```diff
@@ -0,0 +1,354 @@
...
+
+    return ht
+
+
+# Hard-filtered samples
+hard_filtered_samples = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_hard_filtered_samples.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:272]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.hard_filtered_samples.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 23]
```diff
@@ -0,0 +1,354 @@
...
+    },
+)
+
+# Results of running population-based metrics filtering
+# Want to still generate stratified metrics for v4 but will likely use regressed metrics
+stratified_metrics = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_stratified_metrics.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:284]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.stratified_metrics.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 24]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Results of running regressed metrics filtering
+regressed_metrics = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_regressed_metrics.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:295]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.regressed_metrics.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 25]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Ranking of all samples based on quality metrics. Used to remove relateds for PCA.
+pca_samples_rankings = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_pca_samples_ranking.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:306]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.pca_samples_ranking.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 26]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Ranking of all release samples based on quality metrics. Used to remove relateds for release.
+release_samples_rankings = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_release_samples_ranking.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:317]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.release_samples_ranking.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 27]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Duplicated (or twin) samples
+duplicates = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_exomes_v{version}_duplicates.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:328]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.exomes.v{version}.duplicates.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 28]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# PC relate scores for the sample set that overlaps with v3 samples
+v3_v4_pc_relate_pca_scores = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_v3_v{version}_release_pca_scores.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:339]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.v3_v{version}.release_pca_scores.ht"
```

File: gnomad_qc/v4/resources/sample_qc.py [hunk 29]
```diff
@@ -0,0 +1,354 @@
...
+        for version in VERSIONS
+    },
+)
+
+# Relatedness information for the sample set that overlaps with v3 samples
+v3_v4_relatedness = VersionedTableResource(
+    CURRENT_VERSION,
+    {
+        version: TableResource(
+            f"{get_sample_qc_root(version)}/gnomad_v3_v{version}_release_relatedness.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/sample_qc.py:350]:
```suggestion
            f"{get_sample_qc_root(version)}/gnomad.v3_v{version}.release_relatedness.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 30]
```diff
@@ -0,0 +1,240 @@
...
+    :param str truth_sample: Name of the truth sample
+    :param bool mt: Whether path is for a MatrixTable, default is True
+    :return: Path to callset truth sample MT
+    """
+    if mt:
+        return VersionedMatrixTableResource(
+            CURRENT_VERSION,
+            {
+                version: MatrixTableResource(
+                    f"{get_variant_qc_root(version)}/truth_samples/{truth_sample}.mt"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:78]:
```suggestion
                    f"{get_variant_qc_root(version)}/truth_samples/gnomad.exomes.v{version}.{truth_sample}.mt"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 31]
```diff
@@ -0,0 +1,240 @@
...
+                )
+                for version in VERSIONS
+            },
+        )
+    else:
+        return VersionedTableResource(
+            CURRENT_VERSION,
+            {
+                version: TableResource(
+                    f"{get_variant_qc_root(version)}/truth_samples/{truth_sample}.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:88]:
```suggestion
                    f"{get_variant_qc_root(version)}/truth_samples/gnomad.exomes.v{version}.{truth_sample}.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 32]
```diff
@@ -0,0 +1,240 @@
...
+    :param model_id: RF or VQSR model ID for which to return score data.
+    :param bool aggregated: Whether to get the aggregated data.
+         If True, will return the path to Table grouped by bin that contains aggregated variant counts per bin.
+    :return: Path to desired hail Table
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            version: TableResource(
+                f"{get_variant_qc_root(version)}/score_bins/{model_id}.{'aggregated' if aggregated else 'bins'}.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:108]:
```suggestion
                f"{get_variant_qc_root(version)}/score_bins/gnomad.exomes.v{version}.{model_id}.{'aggregated' if aggregated else 'bins'}.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 33]
```diff
@@ -0,0 +1,240 @@
...
+
+    :param model_id: RF or VQSR model ID for which to return score data.
+    :param truth_sample: Which truth sample concordance to analyze (e.g., "NA12878" or "syndip")
+    :return: Path to binned truth data concordance Hail Table
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            version: TableResource(
+                f"{get_variant_qc_root(version)}/binned_concordance/{truth_sample}_{model_id}_binned_concordance.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:128]:
```suggestion
                f"{get_variant_qc_root(version)}/binned_concordance/gnomad.exomes.v{version}.{truth_sample}.{model_id}.binned_concordance.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 34]
```diff
@@ -0,0 +1,240 @@
...
+            - fail_hard_filters - (ht.QD < 2) | (ht.FS > 60) | (ht.MQ < 30)
+
+    :param bool adj: Whether to load 'adj' or 'raw'
+    :return: Table with RF annotations
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            version: TableResource(
+                f"{get_variant_qc_root(version)}/rf/rf_annotations.{'adj' if adj else 'raw'}.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:164]:
```suggestion
                f"{get_variant_qc_root(version)}/rf/gnomad.exomes.v{version}.rf_annotations.{'adj' if adj else 'raw'}.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 35]
```diff
@@ -0,0 +1,240 @@
...
+
+
+def rf_run_path(version: str = CURRENT_VERSION) -> str:
+    """
+    Return the path to the json file containing the RF runs list.
+
+    :param version: Version of RF path to return
+    :return: Path to json file
+    """
+    return f"{get_variant_qc_root(version)}/rf/rf_runs.json"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:178]:
```suggestion
    return f"{get_variant_qc_root(version)}/rf/gnomad.exomes.v{version}.rf_runs.json"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 36]
```diff
@@ -0,0 +1,240 @@
...
+
+def get_rf_model_path(model_id: str, version: str = CURRENT_VERSION) -> str:
+    """
+    Get the path to the RF model for a given run.
+
+    :param model_id: RF run to load
+    :param version: Version of model path to return
+    :return: Path to the RF model
+    """
+    return f"{get_variant_qc_root(version)}/rf/models/{model_id}/rf.model"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:189]:
```suggestion
    return f"{get_variant_qc_root(version)}/rf/models/{model_id}/gnomad.exomes.v{version}.rf.model"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 37]
```diff
@@ -0,0 +1,240 @@
...
+    Get the training data for a given run.
+
+    :param model_id: RF run to load
+    :return: VersionedTableResource for RF training data
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            version: TableResource(
+                f"{get_variant_qc_root(version)}/rf/models/{model_id}/training.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:203]:
```suggestion
                f"{get_variant_qc_root(version)}/rf/models/{model_id}/gnomad.exomes.v{version}.training.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 38]
```diff
@@ -0,0 +1,240 @@
...
+    Get the results of RF filtering for a given run.
+
+    :param model_id: RF run to load
+    :return: VersionedTableResource for RF filtered data
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            version: TableResource(
+                f"{get_variant_qc_root(version)}/rf/models/{model_id}/rf_result.ht"
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:221]:
```suggestion
                f"{get_variant_qc_root(version)}/rf/models/{model_id}/gnomad.exomes.v{version}.rf_result.ht"
```

File: gnomad_qc/v4/resources/variant_qc.py [hunk 39]
```diff
@@ -0,0 +1,240 @@
...
+def final_filter() -> VersionedTableResource:
+    """
+    Get finalized variant QC filtering Table.
+
+    :return: VersionedTableResource for final variant QC data
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
+        {
+            version: TableResource(f"{get_variant_qc_root(version)}/final_filter.ht")
```
Comment (jkgoodrich) [inline gnomad_qc/v4/resources/variant_qc.py:237]:
```suggestion
            version: TableResource(f"{get_variant_qc_root(version)}/gnomad.exomes.v{version}.final_filter.ht")
```

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
Just some more changes for consistency, mostly changing to `.` instead of `_`

**Comment by jkgoodrich** [comment]
Not that it will have a ton in it yet, but lets add a README like we have for v2 and v3 and we should add to it as we add scripts

**Review by jkgoodrich — APPROVED** [review]
LGTM, thanks for moving that and fixing with the hl.vds.VariantDataset

**Pull request merged**